### PR TITLE
Update i2c_gpio.c

### PR DESCRIPTION
**Pull request by CassiniaF**
Fix bug#29308 
let i2c init at POST_KERNEL
sda gpio should init as open drain.

### CODE REVIEW

File: drivers/i2c/i2c_gpio.c [hunk 1]
```diff
@@ -142,7 +142,7 @@ static int i2c_gpio_init(const struct device *dev)
 	}
 
 	err = gpio_config(context->sda_gpio, config->sda_pin,
-			  config->sda_flags | GPIO_INPUT | GPIO_OUTPUT_HIGH);
+			  config->sda_flags | GPIO_OPEN_DRAIN | GPIO_OUTPUT_HIGH);
```
Comment (erwango) [inline drivers/i2c/i2c_gpio.c:145]: This is not related to the PR description. Please remove this

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this fix.
Aside from the comment in the changes, you should fix the commit message, which should looks like:
```
drivers/i2c: i2c_gpio: Title

Description

Signed-off-by: 
```
You can look at commits in zephyr tree for more examples

**Comment by CassiniaF** [comment]
close this one
